pkg/frontend/v2/frontend_scheduler_adapter_test.go

```diff
@@ -0,0 +1,130 @@
+	return instantHTTPReq
+}
+
+const labelValuesURLFormat = "/prometheus/api/v1/label/__name__/values?end=%d&start=%d"
+
+func makeLabelValuesHTTPRequest(ctx context.Context, start, end time.Time) *http.Request {
+	labelValuesURL := fmt.Sprintf(labelValuesURLFormat, end.Unix(), start.Unix())
+	labelValuesHTTPReq, _ := http.NewRequestWithContext(ctx, "GET", labelValuesURL, bytes.NewReader([]byte{}))
+	labelValuesHTTPReq.RequestURI = labelValuesHTTPReq.URL.RequestURI()
+	return labelValuesHTTPReq
+}
+
+func TestExtractAdditionalQueueDimensions(t *testing.T) {
+	adapter := &frontendToSchedulerAdapter{
+		cfg:    Config{QueryStoreAfter: 12 * time.Hour},
+		limits: limits{queryIngestersWithin: 13 * time.Hour},
+	}
+
+	now := time.Now()
+
+	t.Run("query with start after query store after is ingesters only", func(t *testing.T) {
+		ctx := user.InjectOrgID(context.Background(), "tenant-0")
+		startAfterQueryStoreAfter := now.Add(-adapter.cfg.QueryStoreAfter).Add(1 * time.Minute)
+		end := now
+
+		rangeHTTPReq := makeRangeHTTPRequest(ctx, startAfterQueryStoreAfter, end, 60)
+		instantHTTPReq := makeInstantHTTPRequest(ctx, startAfterQueryStoreAfter)
+		labelValuesHTTPReq := makeLabelValuesHTTPRequest(ctx, startAfterQueryStoreAfter, end)
+
+		reqs := []*http.Request{rangeHTTPReq, instantHTTPReq, labelValuesHTTPReq}
+
+		for _, req := range reqs {
+			httpgrpcReq, err := httpgrpc.FromHTTPRequest(req)
+			require.NoError(t, err)
+
+			additionalQueueDimensions, err := adapter.extractAdditionalQueueDimensions(
+				ctx, httpgrpcReq, now,
+			)
+			require.NoError(t, err)
+			require.Equal(t, []string{ShouldQueryIngestersQueueDimension}, additionalQueueDimensions)
+		}
+	})
+
+	t.Run("query with end before query ingesters within is store-gateways only", func(t *testing.T) {
+		ctx := user.InjectOrgID(context.Background(), "tenant-0")
+		startBeforeQueryIngestersWithin := now.Add(-adapter.limits.QueryIngestersWithin("")).Add(-1 * time.Hour)
+		endBeforeQueryIngestersWithin := now.Add(-adapter.limits.QueryIngestersWithin("")).Add(-1 * time.Minute)
+
+		rangeHTTPReq := makeRangeHTTPRequest(ctx, startBeforeQueryIngestersWithin, endBeforeQueryIngestersWithin, 60)
+		instantHTTPReq := makeInstantHTTPRequest(ctx, startBeforeQueryIngestersWithin)
+		labelValuesHTTPReq := makeLabelValuesHTTPRequest(ctx, startBeforeQueryIngestersWithin, endBeforeQueryIngestersWithin)
+
+		reqs := []*http.Request{rangeHTTPReq, instantHTTPReq, labelValuesHTTPReq}
+
+		for _, req := range reqs {
+			httpgrpcReq, err := httpgrpc.FromHTTPRequest(req)
+			require.NoError(t, err)
+
+			additionalQueueDimensions, err := adapter.extractAdditionalQueueDimensions(
+				ctx, httpgrpcReq, now,
+			)
+			require.NoError(t, err)
+			require.Equal(t, []string{ShouldQueryStoreGatewayQueueDimension}, additionalQueueDimensions)
+		}
+	})
+
+	t.Run("query with time between query ingesters within and query store after is ingesters and store-gateways", func(t *testing.T) {
```

Comment:
There are a couple of different cases where we'd query both - it'd be good to see tests for all of them.

I imagine the different cases on a timeline like this, I think the current test case covers case 4:

```
Query ingesters:                  |-------------------
Query store-gateways:   -------------------|

Case 1:                        |--------------|
Case 2:                             |----|
Case 3:                        |---------|
Case 4:                             |---------|
```

Reply:
added and used this illustration in the test comments, thanks! There's some things words just cannot get across with these time ranges, especially with our inconsistent config parameter naming.